### ISSUE TITLE
[M1 Mac] qemu: uncaught target signal 4 (Illegal instruction)

### ISSUE THREAD

**Issue by khirotaka** [docker/for-mac#5446]
When running the [PyTorch Docker image](https://hub.docker.com/r/pytorch/pytorch) on the M1 Mac and executing the code described below, the following error occurred and the interpreter was killed.

```sh
qemu: uncaught target signal 4 (Illegal instruction) - core dumped
Illegal instruction
```

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:  
  `F41CDD94-AAA8-4F6A-AAF1-67EC739038A2/20210311015056`

### Expected behavior
When the following neural net training code is executed, the progress bar will be updated and the accuracy of the trained model will be output.

```python
# test.py
import torch
import torch.nn as nn
import torch.optim as optim
from torch.utils.data import TensorDataset, DataLoader
from tqdm.auto import tqdm

torch.manual_seed(0)


def fetch_dataset():
    x_train = torch.randn(500, 1, 100)
    x_test = torch.randn(500, 1, 100)
    y_train = torch.randint(0, 4, (500, ))
    y_test = torch.randint(0, 4, (500, ))
    return (x_train, y_train), (x_test, y_test)


class CNN(nn.Module):
    def __init__(self, in_features: int, mid_features: int, n_classes: int):
        super(CNN, self).__init__()
        self.conv = nn.Sequential(
            nn.Conv1d(in_features, mid_features, kernel_size=5),
            nn.LeakyReLU(),
            nn.Conv1d(mid_features, mid_features, kernel_size=5),
            nn.LeakyReLU(),
        )
        self.pool = nn.AdaptiveMaxPool1d(1)
        self.fc = nn.Sequential(
            nn.Linear(mid_features, mid_features),
            nn.LeakyReLU(),
            nn.Linear(mid_features, n_classes)
        )

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        batch_size = x.shape[0]
        out = self.conv(x)
        out = self.pool(out)
        out = out.reshape(batch_size, -1)
        out = self.fc(out)
        
        return out


def train(x_train: torch.Tensor, y_train: torch.Tensor, epochs: int) -> CNN:
    in_features = x_train.shape[1]
    n_classes = len(y_train.unique())
    
    train_ds = TensorDataset(x_train, y_train)
    train_loader = DataLoader(train_ds, batch_size=128, shuffle=True)

    model = CNN(in_features, mid_features=64, n_classes=n_classes)
    optimizer = optim.Adam(model.parameters())
    criterion = nn.CrossEntropyLoss()

    model.train()
    with tqdm(range(epochs)) as pbar:
        for epoch in pbar:
            for batch in train_loader:
                data, labels = batch
    
                optimizer.zero_grad()
                out = model(data)
                loss = criterion(out, labels)
                loss.backward()
                optimizer.step()
            
            pbar.set_description("Epoch: {} | Loss: {:.4f}".format(epoch + 1, loss.detach().item()))
            pbar.update(1)

    return model


def evaluate(model: CNN, x_test: torch.Tensor, y_test: torch.Tensor):
    test_ds = TensorDataset(x_test, y_test)
    test_loader = DataLoader(test_ds, batch_size=128, shuffle=False)

    total = 0.0
    correct = 0.0

    model.eval()
    with torch.no_grad():
        for batch in test_loader:
            data, labels = batch
            out = model(data)
            _, predicted = torch.max(out, dim=1)
            total += labels.shape[0]
            correct += (predicted == labels).sum().item()

    print("Accuracy of CNN: {:.4f}".format(correct / total))


def main():
    (x_train, y_train), (x_test, y_test) = fetch_dataset()
    model = train(x_train, y_train, epochs=50)
    evaluate(model, x_test, y_test)


main()
```

and 

```sh
$ python test.py
Epoch: 50 | Loss: 0.5521: 100% 50/50 [00:06<00:00, 7.25it/s]

Accuracy of CNN: 0.2660
```

### Actual behavior
In container

```sh
root@49a19f19df37:/workspace# python test.py 
  0%|               | 0/50 [00:00<?, ?it/s]qemu: uncaught target signal 4 (Illegal instruction) - core dumped
Illegal instruction
```

### Information

  - macOS Version: 11.2.3
  - Docker Desktop Version: 3.1.0(60984)
  - CPU: Apple M1

### Steps to reproduce the behavior

  1. `docker run -it --rm --name pytorch pytorch/pytorch:1.7.1-cuda11.0-cudnn8-devel`
  2. Run the above Python code.

**Comment by stephen-turner** [comment]
There's nothing we can do if qemu can't run the container. This is described in https://docs.docker.com/docker-for-mac/apple-m1/#known-issues where we recommend running arm64 containers on M1 chips.
